### PR TITLE
Feature/systemuserpdp

### PR DESCRIPTION
**Pull request by TheTechArch**
Added Support for SystemUser in PDP

### CODE REVIEW

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs [hunk 1]
```diff
@@ -187,12 +199,70 @@ private static List<XacmlJsonAttribute> CreateSubjectAttributes(IEnumerable<Clai
...
+                }
+                else if (IsPartyIdClaim(claim.Type))
+                {
+                    partyIdAttribute = CreateXacmlJsonAttribute(AltinnXacmlUrns.PartyAttribute, claim.Value, DefaultType, claim.Issuer);
+                }
+                else if (IsResourceClaim(claim.Type))
+                {
+                    partyIdAttribute = CreateXacmlJsonAttribute(AltinnXacmlUrns.ResourceId, claim.Value, DefaultType, claim.Issuer);
+                }
+                else if (IsOrgAttributeClaim(claim.Type))
```
Comment (jonkjetiloye) [inline src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs:222]: OrgAttribute betyr for meg: `urn:altinn:org`
så ungå helst org forkortelsen når det menes organizationnumber

**Comment by TheTechArch** [comment]
Checked scenarios

## When authenticated based on ASPXAUTH (from Altinn 2)
- PartyID is set
- UserID is set

With this change, only userID will be sent to PDP
Code in this method

https://github.com/Altinn/altinn-authentication/blob/main/src/Authentication/Controllers/AuthenticationController.cs#L1029

## Token based on exchange of ID-porten token
- PartyID is set
- UserID is set

With this change, only userID will be sent to PDP

https://github.com/Altinn/altinn-authentication/blob/main/src/Authentication/Controllers/AuthenticationController.cs#L608

## Maskinporten token just exchange
Kun denne som settes
public const string OrgNumber = "urn:altinn:orgNumber";
https://github.com/Altinn/altinn-authentication/blob/main/src/Authentication/Controllers/AuthenticationController.cs#L476

## Maskinporten med exchange av virksomhetsbruker
- PartyID is set
- UserID is set
-  OrgNumber = "urn:altinn:orgNumber";


## Systembrukertoken exchanged
Vil i praksis kun sette orgnummer og er ikke lenger et systembrukertoken
-  OrgNumber = "urn:altinn:orgNumber";

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs (hunk 1 above)
Comment (TheTechArch) [reply src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs]: Agree. Fixed

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs [hunk 2]
```diff
@@ -172,12 +174,22 @@ private static List<XacmlJsonAttribute> CreateSubjectAttributes(IEnumerable<Clai
         {
             List<XacmlJsonAttribute> attributes = new List<XacmlJsonAttribute>();
 
+            XacmlJsonAttribute userIdAttribute = null;
+            XacmlJsonAttribute personUuidAttribute = null;
+            XacmlJsonAttribute partyIdAttribute = null;
+            XacmlJsonAttribute resourceIdAttribute = null;
+            XacmlJsonAttribute orgLegacy = null;
```
Comment (jonkjetiloye) [inline src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs:181]:
```
            XacmlJsonAttribute orgLegacy = null;
            XacmlJsonAttribute orgAttribute = null;
```
bør da også spesifiseres for å ikke kunne forvirres med Org kode.

**Review by jonkjetiloye — APPROVED** [review]
🤞

**Pull request merged**